scipy/sparse/linalg/_propack/readhb/creadhb.f

```diff
@@ -0,0 +1,108 @@
+ccccccccccccccccccccc Read ncol, nnz from HB file cccccccccccccccccccccc
```

Comment:
You answered the question about the `pyf`s in #21, but should these go in the submodule?

Reply:
It's an interface I wrote that doesn't exist in PROPACK-proper needed for loading files programmatically for testing, so my first thought is no, it should stay here.  But if you think it will be easier to maintain in the submodule, I'm happy to move it

Reply:
Oops, didn't remember that you wrote the Fortran yourself. Why write in Fortran and wrap with f2py, though?

Reply:
Past-me probably did this because scipy.io lacks complex HB file format support, there was no library to link to (PROPACK doesn't have file format support in the main libraries, only in the command line interfaces for examples), and all the code existed in the examples, but needed to be reorganized a little bit to do the right thing.  So these files are just me reorganizing some of the existing Fortran code from the examples into proper reusable functions